osu.Game/Tests/Visual/SkinnableTestScene.cs

```diff
@@ -107,12 +111,10 @@ private Drawable createProvider(Skin skin, Func<ISkin, Drawable> creationFunctio
                         Children = new Drawable[]
                         {
                             outlineBox = new OutlineBox(),
-                            (mainProvider = new SkinProvidingContainer(skin)).WithChild(
-                                skinProvider = new SkinProvidingContainer(Ruleset.Value.CreateInstance().CreateLegacySkinProvider(mainProvider, beatmap))
```

Comment:
What does the removal of one of these providers have to do with this change? Isn't throwing in the transformers enough?

Reply:
Skins with a transformer applied should not fall back into themselves again. That eventually results in an `UnsupportedComponentException` when the transformed one returns `null`, as it attempts to lookup from the raw one above it, instead of actually falling back to `SkinManager` or otherwise.